x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/ApiKeyFieldNameTranslators.java

```diff
@@ -45,6 +54,25 @@ public static String translate(String fieldName) {
         throw new IllegalArgumentException("Field [" + fieldName + "] is not allowed for API Key query");
     }
 
+    /**
+     * Translates a query level field name pattern to the matching index level field names.
+     * The result can be the empty set, if the pattern doesn't match any of the allowed index level field names.
+     * If the pattern is actually a concrete field name rather than a pattern,
+     * it is also translated, but only if the query level field name is allowed, otherwise an exception is thrown.
+     */
+    public static Set<String> translatePattern(String fieldNameOrPattern) {
+        Set<String> indexFieldNames = new HashSet<>();
+        for (FieldNameTranslator translator : FIELD_NAME_TRANSLATORS) {
+            if (translator.supports(fieldNameOrPattern)) {
+                indexFieldNames.add(translator.translate(fieldNameOrPattern));
+            }
+        }
+        if (indexFieldNames.isEmpty() && Regex.isSimpleMatchPattern(fieldNameOrPattern) == false) {
+            throw new IllegalArgumentException("Field [" + fieldNameOrPattern + "] is not allowed for API Key query");
```

Comment:
Hmmm, now that you're mentioning it....
I verified that the `simple_query_string` for the regular `_search` endpoints just ignores unrecognized fields (if querying on a unrecognized field, no matches are returned) - and there's no parameter to change this behavior.
So, I believe that you're right that it would probably be more appropriate to also ignore unrecognized fields for `simple_query_string`s for the `/_security/_query/api_key` endpoint, even if all the other allowed query types complain about it.
@n1v0lg Do you have preference here?

Reply:
Pushed [f542fc5](https://github.com/elastic/elasticsearch/pull/104132/commits/f542fc504a86e3111ea8987de06c92a566bba760) to make the SimpleQueryString preprocessing ignore unknown field names, analogously to the the core's processing of the same query type.
The result of it is that a simple query string issued to the `POST /_security/_query/api_key` endpoint behaves identical to when issued to the regular `_search` endpoints (ignores unknown fields).
@n1v0lg If you're leaning towards the less lenient, original implementation, please pitch in, I can easily revert it.

Reply:
Sorry for the delay! I'm ++ on the consistent behavior with regular `_search`. I'll review the rest of the PR shortly.